src/citrine/resources/team.py

```diff
@@ -0,0 +1,293 @@
+        self.actions: List[TEAM_ACTIONS] = actions
+
+    def __str__(self):
+        return '<TeamMember {!r} can {!s} of {!r}>' \
+            .format(self.user.screen_name, self.actions, self.team.name)
+
+
+class Team(Resource['Team']):
+    """
+    A Citrine Team.
+
+    A Team is a collection of projects and resources.
+
+    Parameters
+    ----------
+    name: str
+        Name of the team.
+    description: str
+        Long-form description of the team.
+    session: Session, optional
+        The Citrine session used to connect to the database.
+
+    Attributes
+    ----------
+    uid: UUID
+        Unique uuid4 identifier of this team.
+    created_at: int
+        Time the team was created, in seconds since epoch.
+    name: str
+        Name of the Team
+    description: str
+        Description of the Team
+
+    """
+
+    _response_key = 'team'
+    _resource_type = ResourceTypeEnum.TEAM
+    _api_version = "v3"
+
+    name = properties.String('name')
+    description = properties.Optional(properties.String(), 'description')
+    uid = properties.Optional(properties.UUID(), 'id')
+    created_at = properties.Optional(properties.Datetime(), 'created_at')
+
+    def __init__(self,
+                 name: str,
+                 *,
+                 description: str = "",
+                 session: Optional[Session] = None):
+        self.name: str = name
+        self.description: str = description
+        self.session: Session = session
+
+    def __str__(self):
+        return '<Team {!r}>'.format(self.name)
+
+    def _path(self):
+        return format_escaped_url('/teams/{team_id}', team_id=self.uid)
+
+    def list_members(self) -> List[TeamMember]:
+        """
+        List all of the members in the current team.
+
+        Returns
+        -------
+        List[TeamMember]
+            The members of the current team
+
+        """
+        members = self.session.get_resource(self._path() + "/users",
+                                            version=self._api_version)["users"]
+        return [TeamMember(user=User.build(m), team=self, actions=m["actions"]) for m in members]
+
+    def remove_user(self, user_id: Union[str, UUID]) -> bool:
+        """
+        Remove a User from a Team.
+
+        Parameters
+        ----------
+        user_id: str or uuid
+            The id of the user to remove from the team
+
+        Returns
+        -------
+        bool
+            Returns ``True`` if user successfully removed
+
+        """
+        self.session.checked_post(self._path() + "/users/batch-remove",
+                                  json={"ids": [str(user_id)]}, version=self._api_version)
+        return True  # note: only get here if checked_post doesn't raise error
+
+    def add_user(self, user_id: Union[str, UUID], *,
+                 actions: Optional[List[TEAM_ACTIONS]] = None) -> bool:
+        """
+        Add a User to a Team.
+
+        Adds User with ``READ`` action to the Team.
```

Comment:
Nitpick: Not if the caller provides other actions, as noted below. The caller doesn't need to call update_user_action if they use that parameter.

Reply:
I'll fix this docstring in the followup pr